gpii/node_modules/orca/orcaSettingsHandler.js

```diff
@@ -78,104 +53,165 @@
+            get: {
+                funcName: "gpii.orca.settingsHandler.get",
+                args: ["{that}", "{arguments}.0"]
+            },
+            set: {
+                funcName: "gpii.orca.settingsHandler.set",
+                args: ["{that}", "{arguments}.0"]
+            },
+            getImpl: {
+                funcName: "gpii.orca.settingsHandler.getImpl",
+                args: ["{that}", "{arguments}.0"]
+            },
+            setImpl: {
+                funcName: "gpii.orca.settingsHandler.setImpl",
+                args: ["{that}", "{arguments}.0"]
+            },
+            readSettingsFile: {
+                funcName: "gpii.orca.settingsHandler.readSettingsFile",
+                args: "{that}"
+            }
+        }
+    });
 
-            var change = {
-                path: settingKey,
-                value: settingVal
-            };
-            change.type = settingVal === undefined ? "DELETE" : "ADD";
-            applier.fireChangeRequest(change);
+    gpii.orca.settingsHandler.readSettingsFile = function (that) {
+        var userSettings = fs.readFileSync(orcaSettingsFile);
+        that.applier.change("settings", {});
+        that.applier.change("settings", JSON.parse(userSettings));
+    };
 
-            newSettingsResponse[settingKey] = {
-                "oldValue": oldValue,
-                "newValue": settingVal
-            };
+    gpii.orca.settingsHandler.makeConfigurable = function () {
+        var orcaSpawn = spawn("orca", ORCA_ARGS);
+        var pass = 0;
+        var maxPass = 10;
+        var err;
+
+        while (!fs.existsSync(orcaSettingsFile)) {
+            if (pass === maxPass) {
+                err = "Time limit exceeded [" + maxPass * 500 +
+                      "ms] for creating Orca's configuration file";
+                break;
+            }
+            wait(500);
+            pass++;
+        }
 
-            fluid.set(currentSettingsInProfile, settingKey, settings[settingKey],
-                      fluid.model.escapedSetConfig);
-        });
+        orcaSpawn.kill("SIGKILL");
+        return err;
+    };
 
-        userSettings.profiles[userSettings.general.activeProfile[PROFILE_ID]] = currentSettingsInProfile;
+    gpii.orca.settingsHandler.getImpl = function (that, payload) {
+        var togo = fluid.promise();
+        var exists = fs.existsSync(orcaSettingsFile);
 
-        fs.writeFileSync(orcaSettingsFile,
-                         JSON.stringify(userSettings, null, 4));
+        var getSettings = function () {
+            that.readSettingsFile();
+            var profile = that.model.settings.general.activeProfile[PROFILE_ID];
+            var results = fluid.transform(payload.settings, function (value, key) {
+                var currentValue = fluid.get(that.model.settings, ["profiles", profile, key]);
+                return currentValue;
+            });
+
+            return results;
+        };
+
+        if (exists) {
+            try {
```

Comment:
Very repetitive. This logic should be captured in the getSettings method itself.
Also, the standard form for a rejection payload is
{
    isError: true,
    message: "Couldn't get settings: " + err
}